quinn-udp/src/windows.rs

```diff
@@ -154,4 +334,72 @@ impl UdpSocketState {
     }
 }
 
+fn set_socket_option(
+    socket: &impl AsRawSocket,
+    level: i32,
+    name: i32,
+    value: u32,
+) -> Result<(), io::Error> {
+    let rc = unsafe {
+        WinSock::setsockopt(
+            socket.as_raw_socket() as usize,
+            level,
+            name,
+            &value as *const _ as _,
+            mem::size_of_val(&value) as _,
+        )
+    };
+
+    match rc == 0 {
+        true => Ok(()),
+        false => Err(io::Error::last_os_error()),
+    }
+}
+
 pub(crate) const BATCH_SIZE: usize = 1;
+// Enough to store max(IP_PKTINFO + IP_ECN, IPV6_PKTINFO + IPV6_ECN) bytes (header + data) and some extra margin
+const CMSG_LEN: usize = 128;
+const OPTION_ON: u32 = 1;
+
+// FIXME this could use [`std::sync::OnceLock`] once the MSRV is bumped to 1.70 and upper
+static WSARECVMSG_PTR: Lazy<WinSock::LPFN_WSARECVMSG> = Lazy::new(|| {
+    let s = unsafe { WinSock::socket(WinSock::AF_INET as _, WinSock::SOCK_DGRAM as _, 0) };
+    if s == WinSock::INVALID_SOCKET {
+        tracing::debug!("ignoring WSARecvMsg function pointer due to socket creation error");
```

Comment:
nit: it'd be nice to log the actual error here and below, presumably via `io::Error::last_os_error()`.